api/tests/src/search/backend/test_load_opportunities_to_index.py

```diff
@@ -137,12 +137,16 @@ def test_load_opportunities_to_index(
         opportunity_index_alias,
         load_opportunities_to_index,
     ):
-        index_name = "partial-refresh-index-" + get_now_us_eastern_datetime().strftime(
+        index_prefix = "partial-refresh-index"
+        index_name = f"{index_prefix}-" + get_now_us_eastern_datetime().strftime(
             "%Y-%m-%d_%H-%M-%S"
         )
         search_client.create_index(index_name)
         search_client.swap_alias_index(
-            index_name, load_opportunities_to_index.config.alias_name, delete_prior_indexes=True
+            index_prefix,
+            index_name,
+            load_opportunities_to_index.config.alias_name,
+            delete_prior_indexes=True,
         )
```

Comment:
We need to add a few dedicated tests for this.

We should have tests that specifically check each of the following:
* Create a few indexes with a similar name, they should have been cleaned up.
* Other indexes that might exist are not modified